pkg/operator/controller/ingress/status_test.go

```diff
@@ -407,10 +408,28 @@ func TestComputeIngressDegradedCondition(t *testing.T) {
 			expectIngressDegradedStatus: operatorv1.ConditionFalse,
 			expectRequeue:               false,
 		},
+		{
+			name: "default ingress controller, canary check failing",
+			conditions: []operatorv1.OperatorCondition{
+				cond(IngressControllerCanaryCheckSuccessConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Second*-61)),
+			},
+			expectIngressDegradedStatus: operatorv1.ConditionTrue,
+			expectRequeue:               true,
+			icName:                      "default",
+		},
+		{
+			name: "default ingress controller, canary check passing",
+			conditions: []operatorv1.OperatorCondition{
+				cond(IngressControllerCanaryCheckSuccessConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-1)),
+			},
+			expectIngressDegradedStatus: operatorv1.ConditionFalse,
+			expectRequeue:               false,
+			icName:                      "default",
+		},
```

Comment:
Can we add a case which is not the default that would exercise the conditional: 

```go
if icName == manifests.DefaultIngressControllerName {
```

in `computeIngressDegradedCondition()`?

Reply:
The other tests specify an icName of "" so this is technically already being tested.

Reply:
> The other tests specify an icName of "" so this is technically already being tested.

OK, thanks for confirming.